lib/Tool.php

```diff
@@ -289,29 +289,11 @@ public static function getLanguageFlagFile($language)
      */
     public static function getRoutingDefaults()
     {
-        $config = Config::getSystemConfig();
-
-        if ($config) {
-            // system default
-            $routingDefaults = [
-                'controller' => 'Default',
-                'action' => 'default',
-                'module' => PIMCORE_SYMFONY_DEFAULT_BUNDLE
-            ];
-
-            // get configured settings for defaults
-            $systemRoutingDefaults = $config->documents->toArray();
-
-            foreach ($routingDefaults as $key => $value) {
-                if (isset($systemRoutingDefaults['default_' . $key]) && $systemRoutingDefaults['default_' . $key]) {
-                    $routingDefaults[$key] = $systemRoutingDefaults['default_' . $key];
-                }
-            }
+        $container = \Pimcore::getContainer();
```

Comment:
you could move that to a compiler pass, so there would be one less call to \Pimcore::getContainer() ;)

Reply:
You mean the module-bundle mapping? I think we can do this in the extension class too. Than we can use everywhere the paramerter directly and can remove this function. Is this what you think?

Reply:
Thinking about that: I guess you don't really need the getRoutingDefaults function at all. You can load it from the container directly when you need it. The only place you would need to call ```Pimcore::getContainer``` again is the Renderlet, cause they don't support dependency injection.

Reply:
For me it's fine as it is now, I'd keep `getRoutingDefaults()` anyway to avoid possible BC issues.